pkg/test/framework/components/istio/kube.go

```diff
@@ -902,3 +903,35 @@ func genCommonOperatorFiles(ctx resource.Context, cfg Config, workDir string) (i
 
 	return
 }
+
+func resolveHostDomainToIP(hostDomain string) (string, bool, error) {
+	ips, err := net.LookupIP(hostDomain)
+	if err != nil {
+		return "", false, err
+	}
+	// Use the first resolved IP address
+	for _, ip := range ips {
+		if ipv4 := ip.To4(); ipv4 != nil {
+			return ipv4.String(), true, nil
+		}
+	}
+	return "", false, fmt.Errorf("no IPv4 address found for hostname: %s", hostDomain)
+}
+
+// When the Ingress is a domain name (in public cloud), it might take a bit time to make it reachable.
+func WaitForReachableIngress(hostDomain string) (string, error) {
+	result, err := retry.UntilComplete(func() (any, bool, error) {
+		return resolveHostDomainToIP(hostDomain)
+	}, retry.Timeout(1 * time.Minute), retry.BackoffDelay(10 * time.Second))
+
+	if err != nil {
+		return "", err
+	}
+
+	ip, ok := result.(string)
+	if !ok {
+		return "", fmt.Errorf("unexpected result type, expected string got %T", result)
+	}
+
+	return ip, nil
+}
```

Comment:
Not super familiar with this code, but thoughts on putting this in `istio/pkg/test/kube/util.go` and renaming to `WaitUntilReachableIngress`?

Reply:
@Stevenjin8 

Thanks for the suggestion.
Moved under `istio/pkg/test/kube/util.go`.